### PR TITLE
feat: add graphql for knowledgebase application

### LINKED ISSUE #340: Testcase might fail if arcadia-apiserver restart too many times before arcadia-minio datasource is created
Testcase might fail if arcadia-apiserver restart too many times before arcadia-minio datasource is created, as arcadia-minio creation is a post-install action for helm install, use separate issue to track.

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it

In order to support fixed knowledge base applications in the frontend first, we introduce current pull request.
With current pull request, it is possible to create a fixed knowledge base applications via gql, or you can create this CR directly via the `kubectl apply -f config/samples/app_retrievalqachain_knowledgebase.yaml`

#### Which issue(s) this PR fixes

Fixes #340

### CODE REVIEW

**Comment by Abirdcfly** [comment]
@ggservice007 pls review

File: graphql-server/go-server/pkg/application/knowledgebaseapplication.go [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+	apichain "github.com/kubeagi/arcadia/api/app-node/chain/v1alpha1"
+	apiprompt "github.com/kubeagi/arcadia/api/app-node/prompt/v1alpha1"
+	apiretriever "github.com/kubeagi/arcadia/api/app-node/retriever/v1alpha1"
+	"github.com/kubeagi/arcadia/api/base/v1alpha1"
+	"github.com/kubeagi/arcadia/graphql-server/go-server/graph/generated"
+	"github.com/kubeagi/arcadia/graphql-server/go-server/pkg/utils"
+)
+
+var (
+	appSchem = schema.GroupVersionResource{
```
Comment (bjwswang) [inline graphql-server/go-server/pkg/application/knowledgebaseapplication.go:42]: please move this to common package https://github.com/kubeagi/arcadia/blob/main/graphql-server/go-server/pkg/common/schema.go

File: graphql-server/go-server/pkg/application/knowledgebaseapplication.go [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+		return nil, err
+	}
+
+	app := &v1alpha1.Application{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(objApp.UnstructuredContent(), app); err != nil {
+		return nil, err
+	}
+	UpdateTimestamp := &app.CreationTimestamp.Time
+	status := "Pending"
+	if conditions := app.Status.Conditions; len(conditions) > 0 {
```
Comment (bjwswang) [inline graphql-server/go-server/pkg/application/knowledgebaseapplication.go:86]: If  our condition is `ConditionReady`,we can use `GetCondition(ConditionReady)`

**Comment by Abirdcfly** [comment]
Do not merge, I need test in my win

**Review by bjwswang — APPROVED** [review]

**Pull request merged**